dali/kernels/imgproc/geom/remap.h

```diff
@@ -15,24 +15,25 @@
 #ifndef DALI_KERNELS_IMGPROC_GEOM_REMAP_H_
 #define DALI_KERNELS_IMGPROC_GEOM_REMAP_H_
 
+#include <vector>
 #include <optional>
 #include "dali/core/common.h"
 #include "dali/kernels/kernel.h"
 #include "include/dali/core/boundary.h"
 #include "include/dali/core/geom/box.h"
 
-namespace dali {
-namespace kernels {
+namespace dali::kernels::remap {
 
 
 /**
  * API for Remap operation. Remap applies a generic geometrical transformation to an image.
- * @tparam Backend
- * @tparam T
+ * @tparam Backend Storage backend for data.
+ * @tparam T Type of input and output data.
  */
 template <typename Backend, typename T>
 struct RemapKernel {
   using Border = boundary::Boundary<T>;
+  using MapType = typename std::conditional<std::is_same<T, double>::value, double, float>::type;
```

Comment:
Not sure about this. The type in the map is in the coordinates domain and `T` is the data type, so those are not really related to each other in any way.